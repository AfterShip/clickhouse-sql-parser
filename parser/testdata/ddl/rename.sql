-- rename table
RENAME TABLE t1 TO t11;
RENAME TABLE t1 TO t11 ON CLUSTER 'default_cluster';
RENAME TABLE t1 TO t11, t2 TO t22;
RENAME TABLE t1 TO t11, t2 TO t22 ON CLUSTER 'default_cluster';
-- rename dictionary   
RENAME DICTIONARY t1 TO t11;
RENAME DICTIONARY t1 TO t11 ON CLUSTER 'default_cluster';
RENAME DICTIONARY t1 TO t11, t2 TO t22;
RENAME DICTIONARY t1 TO t11, t2 TO t22 ON CLUSTER 'default_cluster';
-- rename database
RENAME DATABASE t1 TO t11;
RENAME DATABASE t1 TO t11 ON CLUSTER 'default_cluster';
RENAME DATABASE t1 TO t11, t2 TO t22;
RENAME DATABASE t1 TO t11, t2 TO t22 ON CLUSTER 'default_cluster';
