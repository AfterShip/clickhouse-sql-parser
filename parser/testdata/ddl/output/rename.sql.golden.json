[
  {
    "RenamePos": 16,
    "StatementEnd": 38,
    "RenameTarget": "TABLE",
    "TargetPairList": [
      {
        "Old": {
          "Database": null,
          "Table": {
            "Name": "t1",
            "QuoteType": 1,
            "NamePos": 29,
            "NameEnd": 31
          }
        },
        "New": {
          "Database": null,
          "Table": {
            "Name": "t11",
            "QuoteType": 1,
            "NamePos": 35,
            "NameEnd": 38
          }
        }
      }
    ],
    "OnCluster": null
  },
  {
    "RenamePos": 40,
    "StatementEnd": 90,
    "RenameTarget": "TABLE",
    "TargetPairList": [
      {
        "Old": {
          "Database": null,
          "Table": {
            "Name": "t1",
            "QuoteType": 1,
            "NamePos": 53,
            "NameEnd": 55
          }
        },
        "New": {
          "Database": null,
          "Table": {
            "Name": "t11",
            "QuoteType": 1,
            "NamePos": 59,
            "NameEnd": 62
          }
        }
      }
    ],
    "OnCluster": {
      "OnPos": 63,
      "Expr": {
        "LiteralPos": 75,
        "LiteralEnd": 90,
        "Literal": "default_cluster"
      }
    }
  },
  {
    "RenamePos": 93,
    "StatementEnd": 126,
    "RenameTarget": "TABLE",
    "TargetPairList": [
      {
        "Old": {
          "Database": null,
          "Table": {
            "Name": "t1",
            "QuoteType": 1,
            "NamePos": 106,
            "NameEnd": 108
          }
        },
        "New": {
          "Database": null,
          "Table": {
            "Name": "t11",
            "QuoteType": 1,
            "NamePos": 112,
            "NameEnd": 115
          }
        }
      },
      {
        "Old": {
          "Database": null,
          "Table": {
            "Name": "t2",
            "QuoteType": 1,
            "NamePos": 117,
            "NameEnd": 119
          }
        },
        "New": {
          "Database": null,
          "Table": {
            "Name": "t22",
            "QuoteType": 1,
            "NamePos": 123,
            "NameEnd": 126
          }
        }
      }
    ],
    "OnCluster": null
  },
  {
    "RenamePos": 128,
    "StatementEnd": 189,
    "RenameTarget": "TABLE",
    "TargetPairList": [
      {
        "Old": {
          "Database": null,
          "Table": {
            "Name": "t1",
            "QuoteType": 1,
            "NamePos": 141,
            "NameEnd": 143
          }
        },
        "New": {
          "Database": null,
          "Table": {
            "Name": "t11",
            "QuoteType": 1,
            "NamePos": 147,
            "NameEnd": 150
          }
        }
      },
      {
        "Old": {
          "Database": null,
          "Table": {
            "Name": "t2",
            "QuoteType": 1,
            "NamePos": 152,
            "NameEnd": 154
          }
        },
        "New": {
          "Database": null,
          "Table": {
            "Name": "t22",
            "QuoteType": 1,
            "NamePos": 158,
            "NameEnd": 161
          }
        }
      }
    ],
    "OnCluster": {
      "OnPos": 162,
      "Expr": {
        "LiteralPos": 174,
        "LiteralEnd": 189,
        "Literal": "default_cluster"
      }
    }
  },
  {
    "RenamePos": 216,
    "StatementEnd": 243,
    "RenameTarget": "DICTIONARY",
    "TargetPairList": [
      {
        "Old": {
          "Database": null,
          "Table": {
            "Name": "t1",
            "QuoteType": 1,
            "NamePos": 234,
            "NameEnd": 236
          }
        },
        "New": {
          "Database": null,
          "Table": {
            "Name": "t11",
            "QuoteType": 1,
            "NamePos": 240,
            "NameEnd": 243
          }
        }
      }
    ],
    "OnCluster": null
  },
  {
    "RenamePos": 245,
    "StatementEnd": 300,
    "RenameTarget": "DICTIONARY",
    "TargetPairList": [
      {
        "Old": {
          "Database": null,
          "Table": {
            "Name": "t1",
            "QuoteType": 1,
            "NamePos": 263,
            "NameEnd": 265
          }
        },
        "New": {
          "Database": null,
          "Table": {
            "Name": "t11",
            "QuoteType": 1,
            "NamePos": 269,
            "NameEnd": 272
          }
        }
      }
    ],
    "OnCluster": {
      "OnPos": 273,
      "Expr": {
        "LiteralPos": 285,
        "LiteralEnd": 300,
        "Literal": "default_cluster"
      }
    }
  },
  {
    "RenamePos": 303,
    "StatementEnd": 341,
    "RenameTarget": "DICTIONARY",
    "TargetPairList": [
      {
        "Old": {
          "Database": null,
          "Table": {
            "Name": "t1",
            "QuoteType": 1,
            "NamePos": 321,
            "NameEnd": 323
          }
        },
        "New": {
          "Database": null,
          "Table": {
            "Name": "t11",
            "QuoteType": 1,
            "NamePos": 327,
            "NameEnd": 330
          }
        }
      },
      {
        "Old": {
          "Database": null,
          "Table": {
            "Name": "t2",
            "QuoteType": 1,
            "NamePos": 332,
            "NameEnd": 334
          }
        },
        "New": {
          "Database": null,
          "Table": {
            "Name": "t22",
            "QuoteType": 1,
            "NamePos": 338,
            "NameEnd": 341
          }
        }
      }
    ],
    "OnCluster": null
  },
  {
    "RenamePos": 343,
    "StatementEnd": 409,
    "RenameTarget": "DICTIONARY",
    "TargetPairList": [
      {
        "Old": {
          "Database": null,
          "Table": {
            "Name": "t1",
            "QuoteType": 1,
            "NamePos": 361,
            "NameEnd": 363
          }
        },
        "New": {
          "Database": null,
          "Table": {
            "Name": "t11",
            "QuoteType": 1,
            "NamePos": 367,
            "NameEnd": 370
          }
        }
      },
      {
        "Old": {
          "Database": null,
          "Table": {
            "Name": "t2",
            "QuoteType": 1,
            "NamePos": 372,
            "NameEnd": 374
          }
        },
        "New": {
          "Database": null,
          "Table": {
            "Name": "t22",
            "QuoteType": 1,
            "NamePos": 378,
            "NameEnd": 381
          }
        }
      }
    ],
    "OnCluster": {
      "OnPos": 382,
      "Expr": {
        "LiteralPos": 394,
        "LiteralEnd": 409,
        "Literal": "default_cluster"
      }
    }
  },
  {
    "RenamePos": 431,
    "StatementEnd": 456,
    "RenameTarget": "DATABASE",
    "TargetPairList": [
      {
        "Old": {
          "Database": null,
          "Table": {
            "Name": "t1",
            "QuoteType": 1,
            "NamePos": 447,
            "NameEnd": 449
          }
        },
        "New": {
          "Database": null,
          "Table": {
            "Name": "t11",
            "QuoteType": 1,
            "NamePos": 453,
            "NameEnd": 456
          }
        }
      }
    ],
    "OnCluster": null
  },
  {
    "RenamePos": 458,
    "StatementEnd": 511,
    "RenameTarget": "DATABASE",
    "TargetPairList": [
      {
        "Old": {
          "Database": null,
          "Table": {
            "Name": "t1",
            "QuoteType": 1,
            "NamePos": 474,
            "NameEnd": 476
          }
        },
        "New": {
          "Database": null,
          "Table": {
            "Name": "t11",
            "QuoteType": 1,
            "NamePos": 480,
            "NameEnd": 483
          }
        }
      }
    ],
    "OnCluster": {
      "OnPos": 484,
      "Expr": {
        "LiteralPos": 496,
        "LiteralEnd": 511,
        "Literal": "default_cluster"
      }
    }
  },
  {
    "RenamePos": 514,
    "StatementEnd": 550,
    "RenameTarget": "DATABASE",
    "TargetPairList": [
      {
        "Old": {
          "Database": null,
          "Table": {
            "Name": "t1",
            "QuoteType": 1,
            "NamePos": 530,
            "NameEnd": 532
          }
        },
        "New": {
          "Database": null,
          "Table": {
            "Name": "t11",
            "QuoteType": 1,
            "NamePos": 536,
            "NameEnd": 539
          }
        }
      },
      {
        "Old": {
          "Database": null,
          "Table": {
            "Name": "t2",
            "QuoteType": 1,
            "NamePos": 541,
            "NameEnd": 543
          }
        },
        "New": {
          "Database": null,
          "Table": {
            "Name": "t22",
            "QuoteType": 1,
            "NamePos": 547,
            "NameEnd": 550
          }
        }
      }
    ],
    "OnCluster": null
  },
  {
    "RenamePos": 552,
    "StatementEnd": 616,
    "RenameTarget": "DATABASE",
    "TargetPairList": [
      {
        "Old": {
          "Database": null,
          "Table": {
            "Name": "t1",
            "QuoteType": 1,
            "NamePos": 568,
            "NameEnd": 570
          }
        },
        "New": {
          "Database": null,
          "Table": {
            "Name": "t11",
            "QuoteType": 1,
            "NamePos": 574,
            "NameEnd": 577
          }
        }
      },
      {
        "Old": {
          "Database": null,
          "Table": {
            "Name": "t2",
            "QuoteType": 1,
            "NamePos": 579,
            "NameEnd": 581
          }
        },
        "New": {
          "Database": null,
          "Table": {
            "Name": "t22",
            "QuoteType": 1,
            "NamePos": 585,
            "NameEnd": 588
          }
        }
      }
    ],
    "OnCluster": {
      "OnPos": 589,
      "Expr": {
        "LiteralPos": 601,
        "LiteralEnd": 616,
        "Literal": "default_cluster"
      }
    }
  }
]